### PR TITLE
podman auto-update/system df: fix ErrNoSuchCtr/Volume race

### PR DESCRIPTION
**Pull request by Luap99**
If a container is removed during the loop we should just ignore it. No
reason to log this as error to the user.

Fixes https://github.com/containers/podman/issues/23279

#### Does this PR introduce a user-facing change?

```release-note
Fixed a race in podman auto-update and podman system df that would cause them to fail with an error if a container was removed at the same time.
```

### CODE REVIEW

**Comment by Luap99** [comment]
@edsantiago PTAL
That should help the parallel PR.

File: pkg/domain/infra/abi/system.go
Comment (edsantiago) [inline pkg/domain/infra/abi/system.go:208]: LGTM, and it passes in #23275 and on my laptop ... but shouldn't `&&` be `||`? Line 215 also?

**Review by edsantiago — APPROVED** [review]

File: pkg/domain/infra/abi/system.go
Comment (Luap99) [reply pkg/domain/infra/abi/system.go:208]: good catch, yes of course, will repush tomorrow 

/hold

**Comment by edsantiago** [comment]
LGTM. @containers/podman-maintainers PTAL

**Review by giuseppe — APPROVED** [review]
/lgtm

**Comment by edsantiago** [comment]
Ugh, just saw this in one of my laptop parallel runs (in the 610 test):
```
   [05:55:38.197225634] $ /home/esm/src/atomic/2018-02.podman/libpod/bin/podman system df --format {{"\n"}}                                                     
   [05:55:38.297883560] Error: Image 5e5299c48745 exists in local storage but may be corrupted (remove the image to resolve the issue): reading image "5e5299c48
745fd521eedc49358c97e9e8bacfae02c7110e270fad05eaa6d7a05": locating image with ID "5e5299c48745fd521eedc49358c97e9e8bacfae02c7110e270fad05eaa6d7a05": image not k
nown                                                                                                                                                            
   [05:55:38.301820892] [ rc=125 ]
```
Any connection to this PR?

**Comment by Luap99** [comment]
no (I only touched container/volume code) the image query code is in libimage (c/common). It is is possible that it has a similar issue there

**Pull request merged**
